src/Util/FindReflectionOnLine.php

```diff
@@ -23,21 +29,33 @@
      */
     private $sourceLocator;
 
-    public function __construct(SourceLocator $sourceLocator)
+    /**
+     * @var Locator
+     */
+    private $locator;
+
+    public function __construct(SourceLocator $sourceLocator, ?Locator $locator = null)
```

Comment:
IMO, the dependency should be the parser, and the dependency shouldn't be optional.